doc/mgmt/SONiC_PIM_SM_HLD.md

```diff
@@ -0,0 +1,1036 @@
+  
+   1.3 FHR receives the (S,G) join and programs the kernel and the ASIC_DRIVER by following the steps (2 to 9) mentioned in case 1. 
+  Once the (S,G) is programmed the traffic starts to flow natively in FHR and RP.<br> <br>
+    1.4 When LHR receives the multicast data it lifts the data using the programmed (\*,G) and forwards a copy to the receiver. LHR creates (S,G) multicast route with IIF as the RPF interface of the source and OIFs inherited from the (\*,G) route .It forwards the join towards source and it prunes the (\*,G) join towards RP by explicitly sending a (S,G,rpt) prune towards RP.Once the join reaches the FHR the traffic natively flows using the (S,G) join.
+
+#### Case 3: KAT for (S,G) routes
+
+Once the (S,G) routes are learnt and programmed in the hardware, the traffic starts flowing through the data path. To check the liveliness of the (S,G) routes installed in the TIB, the traffic flow for the (S,G) routes in the hardware should be monitored and the status of it should be passed to the Pimd. If the traffic is not flowing for an (S,G) for the default KAT timer of 210 seconds then the entries in the TIB are purged. The flow of(10) to (14) covers this.
+
+10. Once the (S,G) routes are installed in the Hardware, the hardware starts a timer for all the installed (S,G) routes. This timer value is maintained at a lesser value than the original KAT timer and also this timer would be a global timer having the same value across all the VRFs in the hardware.  This value will be updated by FRR once it learns a first ASM route in any VRF and the value would be set to 0 once the last ASM route is deleted across all the VRFs. The hardware KAT timer value is almost 7 times lesser than that of the configured KAT time [say for e.g, if the configured KAT timer value is 210 seconds then the KAT timer in H/W would be 30seconds]. The hardware updates the routes status as and when it changes to DONT_AGE or AGE in the STATE_DB. <br><br>
+    
+11. The ipmcFpmSyncd process will be registered for the event notification for the multicast routes in the state DB. Once the DONT_AGE/AGE status is updated in the DB, ipmcFpmSyncd will receive this event and it will update it's local DB with the age and dont_age counter values.This DB would be updated as and when there is a change in the status of the routes in the Hardware. <br><br>
+    
+12. FRR which runs the KAT timer for each of the routes will query the ipmcFpmsyncd for the routes status. IpmcFpmSyncd will do the route status calculation based on the AGE/DONT_AGE counters and update the status to Pimd.<br><br>
+
+13. Pimd will process this route status and if the route status remains as aged for (2*KAT) value then the routes are purged in the TIB and deleted from Kernel and Hardware too.<br><br>
```

Comment:
Why is this aging 2* KAT? is this from RFC or FRR recommends/behaves like this? Please explain more on this

Reply:
No,it is not a RFC recommendation. This would be the worst case value for a route to be in Software if toggling is seen. Updated the section with the details.